lib/miq_expression.rb

```diff
@@ -1563,37 +1563,44 @@ def self._quick_search?(e)
     end
   end
 
-  def custom_attribute_columns_and_models(expression = nil)
-    return custom_attribute_columns_and_models(exp).uniq if expression.nil?
+  def custom_attribute_columns_and_models
+    @custom_attribute_columns_and_models ||= find_custom_attributes_and_models_recursively || []
+  end
+
+  private
 
+  def find_custom_attributes_and_models_recursively(expression = exp)
     case expression
     when Array
-      expression.flat_map { |x| custom_attribute_columns_and_models(x) }
+      expression.flat_map { |x| find_custom_attributes_and_models_recursively(x) }
```

Comment:
If `find_custom_attributes_and_models_recursively` returns a `nil`, then there will be nils in here
You could add `|| []` in this flat map.

Is it possible to have this method return `[]` when there are no results rather than nil?